### PR TITLE
Add PartitionType and StreamType structs and creation methods

### PR DESCRIPTION
**Pull request by timothycoleman**
Added: PartitionType and StreamType structs and creation methods

### CODE REVIEW

File: src/EventStore.LogV3/RecordCreator.cs [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+		private static byte CurrentVersion(this ref Raw.StreamTypeHeader _) => LogRecordVersion.LogRecordV0;
+
+		private static LogRecordType Type(this ref Raw.EpochHeader _) => LogRecordType.System;
+		private static LogRecordType Type(this ref Raw.PartitionTypeHeader _) => LogRecordType.PartitionType;
+		private static LogRecordType Type(this ref Raw.StreamTypeHeader _) => LogRecordType.StreamType;
+		const int MaxStringBytes = 100;
+		static void PopulateString(string str, Memory<byte> target) {
+			Utf8.FromUtf16(str, target.Span, out _, out var bytesWritten, true, true);
+			if (bytesWritten > MaxStringBytes)
+				throw new ArgumentException($"Name \"{str}s\"is longer than 100 bytes");
```
Comment (shaan1337) [inline src/EventStore.LogV3/RecordCreator.cs:21]:
```suggestion
				throw new ArgumentException($"Name \"{str}\" is longer than 100 bytes");
```

Looks like a small typo and a space is missing

Comment (timothycoleman) [reply src/EventStore.LogV3/RecordCreator.cs:21]: yep good spot!

File: src/EventStore.LogV3/RecordCreator.cs [hunk 2]
```diff
@@ -1,9 +1,30 @@
...
+		private static byte CurrentVersion(this ref Raw.StreamTypeHeader _) => LogRecordVersion.LogRecordV0;
+
+		private static LogRecordType Type(this ref Raw.EpochHeader _) => LogRecordType.System;
+		private static LogRecordType Type(this ref Raw.PartitionTypeHeader _) => LogRecordType.PartitionType;
+		private static LogRecordType Type(this ref Raw.StreamTypeHeader _) => LogRecordType.StreamType;
+		const int MaxStringBytes = 100;
+		static void PopulateString(string str, Memory<byte> target) {
+			Utf8.FromUtf16(str, target.Span, out _, out var bytesWritten, true, true);
+			if (bytesWritten > MaxStringBytes)
+				throw new ArgumentException($"Name \"{str}\" is longer than 100 bytes");
```
Comment (shaan1337) [inline src/EventStore.LogV3/RecordCreator.cs:21]:
```suggestion
				throw new ArgumentException($"Name \"{str}\" is longer than {MaxStringBytes} bytes");
```

@timothycoleman sorry i also just noticed that 100 was hardcoded in the string

**Review by shaan1337 — COMMENTED** [review]
looks good to me, just a minor suggestion :+1:

File: src/EventStore.LogV3/RecordCreator.cs (hunk 2 above)
Comment (timothycoleman) [reply src/EventStore.LogV3/RecordCreator.cs]: oh yeah :+1:

**Review by shaan1337 — APPROVED** [review]

**Review by YoEight — APPROVED** [review]
LGTM

**Pull request merged**